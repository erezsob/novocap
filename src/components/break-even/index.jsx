// @flow
import React, { Component } from 'react';
import Card from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import InfoOutline from 'material-ui-icons/InfoOutline';
import { withStyles } from 'material-ui/styles';
import { BREAK_EVEN_RANGES_TITLES } from '../../constants';
import { getCurrentTimeFrame } from '../../util/time';
import { CircleProgressBar } from '../../components';
import styles from './styles.css';
import jss from './styles';

type Props = {
  classes: {
    card: string,
    iconButton: string
  },
  monthlyCosts: number,
  showModal: Function,
  breakEvenPercent: number,
  timeFrameBreakEven: string,
  fetchRevenueBreakEven: Function
};

const modalSetupObject = {
  modalContentType: 'COSTS_FORM',
  title: 'BREAK EVEN',
  content:
    'Please enter your monthly costs to keep track of your point of break even:',
  label: 'Monthly costs'
};

class BreakEven extends Component<Props> {
  props: Props;

  componentDidMount() {
    const { fetchRevenueBreakEven, timeFrameBreakEven } = this.props;

    fetchRevenueBreakEven({
      ...getCurrentTimeFrame(timeFrameBreakEven),
      periodType: timeFrameBreakEven
    });
  }

  render() {
    const {
      classes: { card, iconButton },
      monthlyCosts,
      showModal,
      breakEvenPercent,
      timeFrameBreakEven
    } = this.props;
    return (
      <Card className={card} onClick={() => showModal(modalSetupObject)}>
        <div className={styles.container}>
          <div className={styles.title}>
            <div>Break Even</div>
            <div className={styles.date}>
              {BREAK_EVEN_RANGES_TITLES[timeFrameBreakEven]}
            </div>
          </div>
          <IconButton aria-label="Info icon" className={iconButton}>
            <InfoOutline />
          </IconButton>
        </div>
        <div className={styles.barContainer}>
          {monthlyCosts ? (
            <CircleProgressBar percent={breakEvenPercent} />
          ) : (
            <div className={styles.addCosts}>Please add costs</div>
          )}
        </div>
      </Card>
    );
  }
}

export default withStyles(jss)(BreakEven);
