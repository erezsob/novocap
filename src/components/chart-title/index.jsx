// @flow
import React from 'react';
import styles from './styles.css';

type Props = {
  label: string
};

export default function ChartTitle({ label }: Props) {
  return <div className={styles.label}>{label}</div>;
}
