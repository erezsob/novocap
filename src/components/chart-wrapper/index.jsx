// @flow
import * as React from 'react';
import styles from './styles.css';

type Props = {
  children?: React.Node
};

export default function ChartWrapper({ children }: Props) {
  return <div className={styles.container}>{children}</div>;
}
