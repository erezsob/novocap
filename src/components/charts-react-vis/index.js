import RadialChart from './radial-chart/index';
import VerticalBarSeries from './vertical-bar-series/index';

export { RadialChart, VerticalBarSeries };
