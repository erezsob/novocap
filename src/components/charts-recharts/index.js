import StackedAreaChart from './stacked-area-chart/index';
import AreaChart from './area-chart/index';

export { StackedAreaChart, AreaChart };
