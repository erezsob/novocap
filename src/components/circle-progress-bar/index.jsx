// @flow
import React from 'react';
import { RadialChart } from 'react-vis';
import { generateData } from '../../util/progress-bar';
import styles from './styles.css';

type Props = {
  percent: number
};

function CircleProgressBar({ percent }: Props) {
  return (
    <div>
      <RadialChart
        width={150}
        height={150}
        innerRadius={60}
        radius={70}
        getAngle={d => d.theta}
        className={styles.chart}
        animation
        colorType="literal"
        data={generateData(percent, styles)}
      />
      <b className={styles.percent}>{`${percent}%`}</b>
    </div>
  );
}

export default CircleProgressBar;
