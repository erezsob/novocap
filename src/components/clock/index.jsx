// @flow
import React, { Component } from 'react';
import { time, date } from './../../util/time';
import styles from './styles.css';

type Props = {};

class Clock extends Component<Props> {
  timer: IntervalID;

  componentDidMount() {
    const clock =
      document && document.getElementsByClassName(`${styles.time}`)[0];
    this.timer = setInterval(() => (clock.innerHTML = time()), 20 * 1000);
  }

  componentWillUnmount() {
    clearInterval(this.timer);
  }

  render() {
    const currentDate = date();
    return (
      <div className={styles.container}>
        <div className={styles.time}>{time()}</div>
        <div className={styles.date}>{currentDate}</div>
      </div>
    );
  }
}

export default Clock;
