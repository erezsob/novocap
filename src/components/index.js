import Clock from './clock';
import CustomListItemText from './custom-list-item-text';
import Header from './header';
import Sidebar from './sidebar';
import Tint from './tint';
import VenueLogo from './venue-logo';
import RightContainer from './right-container';
import Section from './section';
import SectionTitle from './section-title';
import BreakEven from './break-even';
import CircleProgressBar from './circle-progress-bar';
import TimeFrameControls from './time-frame-controls';
import ChartTitle from './chart-title';
import ChartWrapper from './chart-wrapper';
import ItemsPerPage from './items-per-page';
import Paginator from './paginator';
import Datagrid from './table/datagrid';
import TimeCell from './table/time-cell';
import PaymentTypesCell from './table/payments-type-cell';
import MyOBButton from './buttons/my-ob-button/index';
import SubmitButton from './buttons/submit-button';
import CancelButton from './buttons/cancel-button';
import Modal from './modal';
import CustomInputField from './custom-input-field';
import TextFieldWrapper from './text-field-wrapper';
import Footer from './footer';
import Notification from './notification';

export {
  Clock,
  CustomListItemText,
  Header,
  Sidebar,
  VenueLogo,
  Tint,
  RightContainer,
  Section,
  SectionTitle,
  BreakEven,
  CircleProgressBar,
  TimeFrameControls,
  ChartTitle,
  ChartWrapper,
  Paginator,
  ItemsPerPage,
  Datagrid,
  TimeCell,
  PaymentTypesCell,
  MyOBButton,
  SubmitButton,
  CancelButton,
  Modal,
  CustomInputField,
  TextFieldWrapper,
  Footer,
  Notification
};
