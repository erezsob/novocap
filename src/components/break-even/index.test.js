import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BreakEven } from '../../components';

describe('<BreakEven />', () => {
  let wrapper;
  let spyComponentDidMount;
  let spyFetchRevenueBreakEven;

  beforeEach(() => {
    spyComponentDidMount = jest.spyOn(BreakEven.prototype, 'componentDidMount');
    spyFetchRevenueBreakEven = jest.fn();
    wrapper = shallow(
      <BreakEven fetchRevenueBreakEven={spyFetchRevenueBreakEven} />
    ).dive();
  });

  afterEach(() => {
    spyComponentDidMount.mockClear();
    spyFetchRevenueBreakEven.mockClear();
  });

  test('renders', () => {
    expect(wrapper).toHaveLength(1);
  });

  test('renders properly', () => {
    const tree = renderer
      .create(<BreakEven fetchRevenueBreakEven={spyFetchRevenueBreakEven} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Lifecycle methods', () => {
    test('componentDidMount()', () => {
      expect(spyComponentDidMount).toHaveBeenCalledTimes(1);
      expect(spyFetchRevenueBreakEven).toHaveBeenCalledTimes(1);
    });
  });
});
