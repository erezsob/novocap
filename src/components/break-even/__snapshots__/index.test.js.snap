// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreakEven /> renders properly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 BreakEven-card-1"
  onClick={[Function]}
>
  <div
    className="container"
  >
    <div
      className="title"
    >
      <div>
        Break Even
      </div>
      <div
        className="date"
      />
    </div>
    <button
      aria-label="Info icon"
      className="MuiButtonBase-root-36 MuiIconButton-root-30 BreakEven-iconButton-2"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-35"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-39"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-45"
      />
    </button>
  </div>
  <div
    className="barContainer"
  >
    <div
      className="addCosts"
    >
      Please add costs
    </div>
  </div>
</div>
`;
